diesel/src/pg/types/network_address.rs

```diff
@@ -0,0 +1,204 @@
+extern crate ipnetwork;
+extern crate libc;
+
+use std::io::prelude::*;
+use std::error::Error;
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use pg::Pg;
+use types::{self, ToSql, IsNull, FromSql};
+use self::ipnetwork::{IpNetwork,Ipv4Network,Ipv6Network};
+
+#[cfg(windows)]
+const AF_INET: u8 = 2;
+// Maybe not used, but defining to follow Rust's libstd/net/sys
+#[cfg(redox)]
+const AF_INET: u8 = 1;
+#[cfg(not(any(windows, redox)))]
+const AF_INET: u8 = libc::AF_INET as u8;
+
+
+const PGSQL_AF_INET: u8 = AF_INET + 0;
+const PGSQL_AF_INET6: u8 = AF_INET + 1;
+
+primitive_impls!(Cidr -> (IpNetwork, pg: (650, 651)));
+primitive_impls!(Inet -> (IpNetwork, pg: (869, 1041)));
+
+macro_rules! err {
+    () => (return Err("invalid network address format".into()))
+}
+
+macro_rules! assert_or_error {
+    ($cond: expr) => {
```

Comment:
Maybe extend this (and `err!`) to also take an optional message. This way, you could make the user facing errors as well as the asserts below more descriptive.